src/Lykke.Service.Assets.Services/AssetConditionService.cs

```diff
@@ -11,53 +11,251 @@ namespace Lykke.Service.Assets.Services
+
+            IEnumerable<IAssetCondition> defaultLayerConditions
+                = await _assetConditionRepository.GetAsync(assetDefaultLayer.Id);
+
+            var map = new Dictionary<string, AssetCondition>();
+
+            // Initialize asset conditions using default layer conditions
+            foreach (IAssetCondition condition in defaultLayerConditions)
+            {
+                map[condition.Asset] = Mapper.Map<AssetCondition>(condition);
+            }
+
+            // Merge client conditions layers
+            IEnumerable<IAssetConditionLayer> layers = await GetLayersAsync(clientId);
+
+            foreach (IAssetConditionLayer layer in layers.OrderBy(e => e.Priority))
+            {
+                var explicitAssets = new HashSet<string>();
+
+                IEnumerable<IAssetCondition> conditions = await _assetConditionRepository.GetAsync(layer.Id);
+                
+                // Apply explicit assets conditions
+                foreach (IAssetCondition condition in conditions)
                 {
-                    if (!result.TryGetValue(condition.Key, out var value))
-                    {
-                        value = new AssetCondition(condition.Key);
-                        result[condition.Key] = value;
-                    }
-                    value.Apply(condition.Value);
+                    if (!map.TryGetValue(condition.Asset, out var value))
+                        map[condition.Asset] = Mapper.Map<AssetCondition>(condition);
+                    else
+                        value.Apply(condition);
+
+                    explicitAssets.Add(condition.Asset);
+                }
+
+                IAssetConditionSettings defaultAssetCondition = await _assetDefaultConditionRepository.GetAsync(layer.Id);
+
+                if(defaultAssetCondition == null)
+                    continue;
+
+                // Apply implicit assets conditions
+                IEnumerable<string> implicitAssets = map.Keys.Where(o => !explicitAssets.Contains(o));
+
+                foreach (string asset in implicitAssets)
+                {
+                    map[asset].Apply(defaultAssetCondition);
                 }
             }
 
-            return result.ToDictionary(e => e.Key, e => e.Value as IAssetCondition);
+            assetConditions = map.Values.Cast<IAssetCondition>().ToList();
+
+            // Update asset conditions cache
+            await _cacheManager.SaveAssetConditionsForClientAsync(clientId, assetConditions);
+
+            return assetConditions;
         }
 
         public async Task<IAssetConditionLayerSettings> GetAssetConditionsLayerSettingsByClient(string clientId)
         {
-            var layersIds = await _assetConditionLayerLinkClientRepository.GetAllLayersByClientAsync(clientId);
-            var layers = await _assetConditionLayerRepository.GetByIdsAsync(layersIds);
+            IEnumerable<IAssetConditionLayer> layers = await GetClientLayers(clientId);
```

Comment:
Why var is replaced with explicit type?

Reply:
I follow the rule: Don't use var when the return type isn't obvious.

Reply:
Method name is self-descriptive enough to use var here